scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/AvailableConnections.java

```diff
@@ -0,0 +1,112 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Bosch Software Innovations GmbH and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Achim Kraus (Bosch Software Innovations GmbH) - initial implementation
+ ******************************************************************************/
+package org.eclipse.californium.scandium.dtls;
+
+/**
+ * Pair of connections.
+ * 
+ * Helper to prevent duplicated calls to
+ * {@link ResumptionSupportingConnectionStore#find(SessionId)}.
+ * 
+ * <ul>
+ * <li>connection associated with the peer's address or cid</li>
+ * <li>connection associated with the session id</li>
+ * </ul>
+ */
+public final class AvailableConnections {
+
+	/**
+	 * Connection associated with the peer's address. May be {@code null}, if no
+	 * connection is associated with the peer's address.
+	 */
+	private Connection byAddress;
+	/**
+	 * Connection associated with the session id. May be {@code null}, if no
+	 * connection is associated with the session id.
+	 */
+	private Connection bySessionId;
+	/**
+	 * Indicates, that {@link #setConnectionBySessionId(Connection)} has been
+	 * called.
+	 */
+	private boolean setBySessionId;
```

Comment:
Why do we need an attribute ? `bySessionId == null` is not the same meaning ?

Reply:
Let me try to explain:
With the `VerifyPeersOnResumptionThreshold` the "find" will be called potentially from two functions: `isClientInControlOfSourceIpAddress` and `resumeExistingSession`.
With this PR I want to ensure explicit, that `find`is only called once per received client_hello (may be more within the handshake), even if `null`is returned.
There may be "complexer" explanations, why `find` is "anyway" only called once, but for me this seems to be more fragile.

Reply:
Ok I get it.